### PR TITLE
[GH-321] Moved all modules from Structures

### LINKED ISSUE #321: Structures modules reorganisation
We should move all or most of our structures into folders relevant to function. Motivation:
- recently most of our code changes consist of adding more logic to structures. For example code for creating spend transaction is inside `Aecore.Structures.Account` module although it doesn't even interact with any Account structure or any other function inside that module!
- `lib/aecore/structures` has incomparably more modules then any other folder
- `lib/aecore/txs` despite the name contains only txs pool
- general rule is that code should be split into modules according to function, not programming language properties

### PR DESCRIPTION
**Pull request by cytadela8**
Only `Aecore.Chain.Account` is in a little bit miserable place. If somebody has better idea please tell me

This closes #321

commands to reproduce my results:
```
find . -type f -name "*.ex*" -exec sed -i.bak "s/Aecore.Structures.Block/Aecore.Chain.Block/g" {} \;
find . -type f -name "*.ex*" -exec sed -i.bak "s/Aecore.Structures.Header/Aecore.Chain.Header/g" {} \;
find . -type f -name "*.ex*" -exec sed -i.bak "s/Aecore.Structures.Account/Aecore.Chain.Account/g" {} \;
find . -type f -name "*.ex*" -exec sed -i.bak "s/Aecore.Structures.DataTx/Aecore.Tx.DataTx/g" {} \;
find . -type f -name "*.ex*" -exec sed -i.bak "s/Aecore.Structures.SignedTx/Aecore.Tx.SignedTx/g" {} \;
find . -type f -name "*.ex*" -exec sed -i.bak "s/Aecore.Structures.SpendTx/Aecore.Tx.Type.SpendTx/g" {} \;
find . -type f -name "*.ex*" -exec sed -i.bak "s/Aecore.Structures.OracleRegistrationTx/Aecore.Tx.Type.OracleRegistrationTx/g" {} \;
find . -type f -name "*.ex*" -exec sed -i.bak "s/Aecore.Structures.OracleQueryTx/Aecore.Tx.Type.OracleQueryTx/g" {} \;
find . -type f -name "*.ex*" -exec sed -i.bak "s/Aecore.Structures.OracleResponseTx/Aecore.Tx.Type.OracleResponseTx/g" {} \;
find . -type f -name "*.ex*" -exec sed -i.bak "s/Aecore.Structures.OracleExtendTx/Aecore.Tx.Type.OracleExtendTx/g" {} \;
find . -type f -name "*.ex*" -exec sed -i.bak "s/Aecore.Txs/Aecore.Tx/g" {} \;
find . -type f -name "*.ex*" -exec sed -i.bak "s/Aecore.Chain.Account/Aecore.Account.Account/g" {} \;
find . -type f -name "*.ex*" -exec sed -i.bak "s/Aecore.Tx.Type.SpendTx/Aecore.Account.Tx.SpendTx/g" {} \;
find . -type f -name "*.ex*" -exec sed -i.bak "s/Aecore.Tx.Type.Oracle/Aecore.Oracle.Tx.Oracle/g" {} \;
find . -type f -name "*.ex*" -exec sed -i.bak "s/Aecore.Structures.Transaction/Aecore.Tx.Transaction/g" {} \;
find . -type f -name "*.ex*" -exec sed -i.bak "s/Aecore.Structures.AccountStateTree/Aecore.Account.AccountStateTree/g" {} \;
find . -type f -name "*.ex*" -exec sed -i.bak "s/Aecore.Structures.Chainstate/Aecore.Chain.Chainstate/g" {} \;
find -name "*.bak" -delete
```

### CODE REVIEW

**Comment by cytadela8** [comment]
Those commands might be useful when doing a merge.

**Comment by cytadela8** [comment]
@thepiwo We can go with that. 

I have one question then: Where should SpendTx and Account go (currently only Account is in a little bit random place)?

**Comment by d-velev** [comment]
I agree with "I think we should leave feature specific transaction structures in /feature", as for where to put Account, I thought of putting it in Chain, you've done that already so this looks fine to me

**Comment by cytadela8** [comment]
Ok, so if we are putting tx-s in "/future" then where should SpendTx go? To Chain together with Account? Or should it be together with signedtx and datatx. Or maybe we should separate Account and SpendTx into "/account"? (In future some genserver for keeping track of our own account txs, nonce and some other properties could go there.)

**Comment by cytadela8** [comment]
@thepiwo DONE

File: apps/aecore/lib/aecore/tx/transaction.ex [hunk 1]
```diff
@@ -4,8 +4,8 @@ defmodule Aecore.Structures.Transaction do
   child tx of DataTx should implement to work correctly on the blockchain
```
Comment (DanielaIvanova) [inline apps/aecore/lib/aecore/tx/transaction.ex]: Module name is not correct.

File: apps/aecore/lib/aecore/tx/transaction.ex [hunk 2]
```diff
@@ -4,8 +4,8 @@ defmodule Aecore.Structures.Transaction do
...
   """
```
Comment (DanielaIvanova) [inline apps/aecore/lib/aecore/tx/transaction.ex]: Module name is not correct.

Comment (cytadela8) [reply apps/aecore/lib/aecore/tx/transaction.ex]: fixed

**Comment by thepiwo** [comment]
merge this after #317

**Comment by thepiwo** [comment]
@cytadela8 can you resolve the conflicts and merge?

**Comment by cytadela8** [comment]
@thepiwo should be done. Force pushed, because it's easier to redo all the changes then to merge. (I keep the script in PR description up-to-date)

**Pull request merged**